e2e/test/scenarios/dashboard-cards/reproductions/23137-click-behavior-not-working-for-gauge-and-progress.cy.spec.js

```diff
@@ -0,0 +1,76 @@
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import { ORDERS_QUESTION_ID } from "e2e/support/cypress_sample_instance_data";
+import { restore, visitDashboard } from "e2e/support/helpers";
+
+const { REVIEWS_ID } = SAMPLE_DATABASE;
+
+describe("issue 23137", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.intercept("POST", "/api/dataset").as("dataset");
+  });
+
+  it("should navigate to a target from a gauge card (metabase#23137)", () => {
+    const target_id = ORDERS_QUESTION_ID;
+
+    cy.createQuestionAndDashboard({
+      questionDetails: getQuestionDetails({ display: "gauge" }),
+    }).then(({ body: { id, card_id, dashboard_id } }) => {
+      cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+        cards: [getDashcardDetails({ id, card_id, target_id })],
+      });
+
+      visitDashboard(dashboard_id);
+    });
+
+    cy.findByTestId("gauge-arc-1").click();
+    cy.wait("@dataset");
+    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+    cy.findByText("Orders");
+  });
+
+  it("should navigate to a target from a progress card (metabase#23137)", () => {
+    const target_id = ORDERS_QUESTION_ID;
+
+    cy.createQuestionAndDashboard({
+      questionDetails: getQuestionDetails({ display: "progress" }),
+    }).then(({ body: { id, card_id, dashboard_id } }) => {
+      cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+        cards: [getDashcardDetails({ id, card_id, target_id })],
+      });
+
+      visitDashboard(dashboard_id);
+    });
+
+    cy.findByTestId("progress-bar").click();
+    cy.wait("@dataset");
+    // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+    cy.findByText("Orders");
+  });
+});
+
+const getQuestionDetails = ({ display }) => ({
```

Comment:
I'd just create 2 top-level variables maybe
- `GAUGE_QUESTION_DETAILs` and
- `PROGRESS_QUESTION_DETAILS`
Or other names as you see fit as we usually declare our dashboard/question details in the top of the files. And there isn't much duplication here at all. Declaring them at the top would be more consistent.

Reply:
77b5a1fe671686720a9e8b7f8e34e6cf3b90e782